sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/Validations.scala

```diff
@@ -241,6 +243,25 @@ object Validations {
     }
   }
 
+  private def viewMustHaveMethodLevelSubscriptionWhenTransformingUpdates(component: Class[_]): Validation = {
+    if (hasValueEntitySubscription(component)) {
+      val tableType: Class[_] = tableTypeOf(component)
+      val valueEntityClass: Class[_] =
+        component.getAnnotation(classOf[Subscribe.ValueEntity]).value().asInstanceOf[Class[_]]
+      val entityStateClass = valueEntityStateClassOf(valueEntityClass)
+
+      when(entityStateClass != tableType) {
+        val message = s"View subscribes to ValueEntity [${valueEntityClass.getName}] and subscribes to state changes " +
+          s"which will be of type [${entityStateClass.getName}] but view type parameter is [${tableType.getName}] which does not match, " +
+          "the types of the entity and the subscribing must be the same. Make sure that @Subscribe.ValueEntity annotation is used on a method like " +
+          s"`UpdateEffect<${tableType.getName}> onChange(${entityStateClass.getName} state)`."
```

Comment:
I'm not sure users will understand it. I think we can be more direct. Something like: 

"You are using a type level annotation in this View and that requires the View type to match the ValueEntity type [${entityStateClass.getName}]. If your intention is to transform the type, you should instead add a method like `UpdateEffect<${tableType.getName}> onChange(${entityStateClass.getName} state)` and move the `@Subscribe.ValueEntity` to it."

Something on those lines. WDYT?

Reply:
Updated.